src/main/java/de/siegmar/logbackgelf/SimpleJsonEncoder.java

```diff
@@ -107,36 +107,44 @@ private void appendKey(final String key) throws IOException {
      */
     @SuppressWarnings("checkstyle:cyclomaticcomplexity")
     private void escapeString(final String str) throws IOException {
+        int currentIndex = 0;
+        int lastCopyIndex = 0;
         for (final char ch : str.toCharArray()) {
+            String escaped = null;
             switch (ch) {
                 case QUOTE:
                 case '\\':
                 case '/':
-                    writer.append('\\');
-                    writer.append(ch);
+                    escaped = "\\" + ch;
                     break;
                 case '\b':
-                    writer.append("\\b");
+                    escaped = "\\b";
                     break;
                 case '\f':
-                    writer.append("\\f");
+                    escaped = "\\f";
                     break;
                 case '\n':
-                    writer.append("\\n");
+                    escaped = "\\n";
                     break;
                 case '\r':
-                    writer.append("\\r");
+                    escaped = "\\r";
                     break;
                 case '\t':
-                    writer.append("\\t");
+                    escaped = "\\t";
                     break;
-                default:
-                    if (ch < ' ') {
-                        writer.append(escapeCharacter(ch));
-                    } else {
-                        writer.append(ch);
-                    }
             }
+            if (ch < ' ') {
+                escaped = escapeCharacter(ch);
+            }
+            if (escaped != null) {
+                writer.append(str.substring(lastCopyIndex, currentIndex));
```

Comment:
As this method already gets the char array representation of `str` (via `str.toCharArray()`), I assume that using `java.io.Writer#write(char[], int, int)` would be a lot faster than slicing the string. Could you check this?

Reply:
Changed to `Writer.write` instead of append, and it performed way better than the first changes. Also performed better than the upstream version of this code, I will put the reports on the comments bellow